scripts/bullet_script.gd

```diff
@@ -0,0 +1,32 @@
+class_name bullet
+extends Area
+
+var speed = 80
+var life_time = 3
+var my_shooter : KinematicBody
+onready var timer = get_node("Timer")
+
+func _ready():
+	timer.set_wait_time(life_time)
+	timer.start()
+	
+
+func _physics_process(delta):
+	self.transform.origin += self.transform.basis.z * speed * delta
+
+
+func _on_bullet_body_entered(body):
+	if body != my_shooter:
+		queue_free()
+
+
+func _on_Timer_timeout():
+	queue_free()
+
+
+func _on_bullet_area_entered(area):
+	var target = area.get_parent()
+	if target != my_shooter && target.get('damagable_type') != null:
+		if target.damagable_type == "actor":
+			target.take_damage(100,"kinetic")
```

Comment:
Jos tässä tarkistuksessa riittäis kattoa, että `target is Actor`, niin sitten ei IMO tarvis tehä mitään muita tarkistuksia. Ehkä ton damage-tapahtuman vois handlata oikein tuolla `Actor.take_damage` metodissa? Sillon bulletin ei tarvis tietää mitään ylimäärästä. Actorilla vois olla joku `is_invulnerable` tms. flagi, jonka perusteella toi `take_damage` sitten tekee tai ei tee damagea tai muita efektejä (kuten knockbackia), esim. dodgen aikana?

Reply:
Toi eka check on vaan siitä jääny kun bullet pysty osumaan itteensä, ja tokan checkin idea on, et jos osuu johonkin areaan joka ei ymmärrä take_damage(), niin ei kaadu.. Saman voi hoitaa tolla classilla joo, mun idea oli vaan et jos halutaan tehä johonki muuhun kun actoreihin samalla lailla, niin ei tartte kattoo target is Actor || target is DamagableProp || etc. Mut tässä vaiheessa voidaan toki tehä classilla myös